src/app/pages/system/enclosure/components/enclosure-side/enclosure-svg/enclosure-svg.component.ts

```diff
@@ -246,25 +252,65 @@ export class EnclosureSvgComponent implements OnDestroy {
     const prevSlotExists = !!selectedSlot;
 
     if (!isNewSlotEmpty && prevSlotExists && slot.dev === selectedSlot.dev) {
-      this.selectedSlot.set(null);
       return;
     }
 
     this.selectedSlot.set(slot);
   };
 
   private addTint(slot: DashboardEnclosureSlot): void {
-    const overlay = this.overlayRects[slot.drive_bay_number];
+    const tintTargets: NodeListOf<SVGGElement> = this.getTintTarget(slot.drive_bay_number);
+    const slotTint = this.slotTintFn()(slot);
+
+    if (slotTint) {
+      tintTargets.forEach((tintTarget: SVGGElement) => {
+        this.renderer.addClass(tintTarget, 'tinted');
+        this.renderer.setStyle(tintTarget, 'fill', slotTint);
+        this.renderer.setStyle(tintTarget, 'filter', 'brightness(1.25)');
+      });
+      this.dimSlot(slot.drive_bay_number, 1);
+    } else {
+      tintTargets.forEach((tintTarget: SVGGElement) => {
+        this.renderer.removeClass(tintTarget, 'tinted');
+        this.renderer.setStyle(tintTarget, 'fill', null);
+        this.renderer.setStyle(tintTarget, 'filter', 'brightness(1)');
+      });
+      this.dimSlot(slot.drive_bay_number, this.emptyOpacity);
+    }
+  }
 
-    this.renderer.removeClass(overlay, 'tinted');
+  private dimSlot(slotNumber: number, opacity: number): void {
+    const slotId: string = 'DRIVE_CAGE_' + slotNumber.toString();
```

Comment:
I would suggest updating code that initially looks for drives cages in `processSvg` to store them as class properties and then using that here and below.

Reply:
@undsoft Updated the file. Please check again.